shared/naturalcrit/codeEditor/codeEditor.jsx

```diff
@@ -431,9 +433,21 @@ const CodeEditor = createClass({
 				}
 				text = foldPreviewText || `Lines ${from.line+1}-${to.line+1}`;
 
-				text = text.trim();
+
+				text = text.replace('{', '').trim();
+
+				// Extra data url chomping
+				// Try to make it pretty...
+				const startOfData = text.indexOf('data:');
+				if(startOfData) {
+					text = (startOfData > maxLength) ?
+						`${text.slice(0, text.indexOf(':') + 1)} ... ${text.slice(startOfData, startOfData + 5)} ...` :
+						`${text.slice(0, text.indexOf('data:') + 5)} ...`;
+				}
```

Comment:
This section might need a rework. `text.indexOf` will return -1 if there is no match, which is not a falsy value. So the if statement will still run even if `data:` is not present.

This is causing non-`:data:` strings to get truncated aggressively:

##### Unfolded:
![image](https://github.com/user-attachments/assets/5209795d-9450-48e5-863a-4a727cb2ae9f)

##### Folded - Before
![image](https://github.com/user-attachments/assets/1c4c2ffc-2c47-4bf6-aaa4-6e92f331bbf0)

##### Folded - After
![image](https://github.com/user-attachments/assets/6ee9632e-73de-40f2-a442-467f8e89ee46)

Reply:
Wow, instead of using actual css elipsis we just input the ... huh? Interesting choice.

Reply:
I don't think CSS ellipses supports specifying a character limit, or does it?

Reply:
IIRC, it's works on a linear width specified in px, pt, em, etc.
We could thus use the `ch` unit (which is the width of the `0` character).

Our editor uses a monospaced font.

Reply:
There we go, monospaced font + width:5ch =  specified width of 4 characters (because the elipsis will take the space for 1 character..

Reply:
Though we want a width of 50 characters. This PR is truncating to 4 by mistake.

Reply:
I am resolving this solely against the truncation issue as the CSS based ellipses is a larger scoped change.